### DISCUSSION TITLE
Change Vanilla OS to a hypervisor based System

### DISCUSSION THREAD

**Discussion by Mannshoch** [Vanilla-OS/live-iso#355] [Ideas]
Let's dream strange Ideas. 

Example how it could look like:
- Hardware:
  - Grub + Hypervisor (example proxmox)
    - VM Root (readonly IMG)
    - VM or Overlay for user and network management
    (gdm, lightdm, what ever. Takes the login, tells Hypervisor what Root-VM (if more than one) to boot decrypt the user drive and login the user.)
    - Encrypted drives
      - Overlayfs for root (alluser)
      - User 
        - Overlay for root (user if allowed)
        - home
        
Software:
Hypervisor: 
- Auto-Update VM
- Place for driver and settings except User-devices Like most USB devices

User and Network management: 
- Contain for this PC allowed users 
- if existing: Machine based VPN (LDAP)

Communication between VMs -> webhooks

**Comment by axtloss** [comment]
This really is just Qubes OS, and honestly a bit out of scope for vanilla
although some things 
proxmox isn't the type of hypervisor to be used here, it'd add too much overhead, a level 0 hypervisor like Xen (which is what qubes os uses) is better fit here

webhooks aren't the right tool for communicating between vms, as vms may want to be airgapped, which is why communicating over shared memory or in the case of Xen using vchannels (like Qubes OS does for their rpc solution qrexec)
